de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractChangePriceQtyRepository.java

```diff
@@ -0,0 +1,106 @@
+/**
+ *
+ */
+package de.metas.contracts.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.Services;
+import org.springframework.stereotype.Component;
+
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.contracts.model.I_C_SubscriptionProgress;
+import de.metas.contracts.model.X_C_SubscriptionProgress;
+import de.metas.contracts.subscription.ISubscriptionDAO;
+import de.metas.contracts.subscription.ISubscriptionDAO.SubscriptionProgressQuery;
+import de.metas.invoicecandidate.api.IInvoiceCandidateHandlerBL;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@Component
+public class ContractChangePriceQtyRepository
+{
+	public void changePriceIfNeeded(@NonNull final I_C_Flatrate_Term contract, final BigDecimal priceActual)
```

Comment:
i think this is not a repository..
a repositor would just get ythe entities from our DB for you..
we think this is a service
..aand, please use the `@Service` annotation..to make it more transparent..

Reply:
Yes. After reading, I also had that feeling. But wasn't sure. Is the first attempt.
Thx!